autopts/wid/has.py

```diff
@@ -0,0 +1,153 @@
+#
+# auto-pts - The Bluetooth PTS Automation Framework
+#
+# Copyright (c) 2023, Oticon.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms and conditions of the GNU General Public License,
+# version 2, as published by the Free Software Foundation.
+#
+# This program is distributed in the hope it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+
+import logging
+import re
+import sys
+
+from enum import IntFlag
+from autopts.pybtp import btp
+from autopts.ptsprojects.stack import get_stack
+from autopts.pybtp.types import WIDParams
+
+class PresetProperty(IntFlag):
+    BT_HAS_PROP_NONE      = 0x00
+    BT_HAS_PROP_WRITABLE  = 0x01
+    BT_HAS_PROP_AVAILABLE = 0x02
+
+log = logging.debug
+
+
+def has_wid_hdl(wid, description, test_case_name, logs=True):
+    if logs:
+        log(f'{has_wid_hdl.__name__}, {wid}, {description}, {test_case_name}')
+    module = sys.modules[__name__]
+
+    try:
+        handler = getattr(module, f'hdl_wid_{wid}')
+        return handler(WIDParams(wid, description, test_case_name))
+    except AttributeError as e:
+        logging.exception(e)
+
+
+# wid handlers section begin
+
+"""
+    Please add a new Preset Record.
+"""
+def hdl_wid_450(_:WIDParams):
+    n, properties = 4, PresetProperty.BT_HAS_PROP_AVAILABLE | PresetProperty.BT_HAS_PROP_WRITABLE
+    btp.has_add_preset(n, properties, 'PRESET_' + str(n))
+    return True
+
```

Comment:
Pycharm IDE highlights me some warnings about Python PEP 8 standard, which we try to follow (most of the time ;) ):
- comments should be inside the functions
- put 2 blank lines between functions
- put a space after a colon `_: WIDParams`
```suggestion
def hdl_wid_450(_: WIDParams):
    """
    Please add a new Preset Record.
    """
    n, properties = 4, PresetProperty.BT_HAS_PROP_AVAILABLE | PresetProperty.BT_HAS_PROP_WRITABLE
    btp.has_add_preset(n, properties, 'PRESET_' + str(n))
    return True


```

Reply:
Done.